templates/solidity-frameworks/hardhat/packages/hardhat/hardhat.config.ts.template.mjs

```diff
@@ -25,6 +25,7 @@ const etherscanApiKey = process.env.ETHERSCAN_API_KEY || "DNXJA8RX2Q3VZ4URQIWP7Z
 const config: HardhatUserConfig = {
   solidity: {
     compilers: [
+      ${compilers[0] && `${compilers[0]},`}
```

Comment:
I think here is a nice discussion so in some cases (like this case you need to pass things that are elements of parent object / array. 

The corresponding `.args.mjs` looks like this: 

![image](https://github.com/user-attachments/assets/92266861-3b00-474f-b486-01aecfa2d59a)


CC @rin-st , @carletex how do you envision this? 

I think let people pass that objects in array? 

```js
const compilers = [{
 version: "0.8.17",
 settings: {}
}, {
 version: "0.8.17",
 settings: {},
}]
```

And then in `.template.mjs` we do this : 


```js
const massagedCompilers = JSON.stringify(compilers[0]);

return  `{
  compilers: [
    ${compilers[0]} ? `...${massagedCompilers},` : ''
  ]
}`
```

The small down side of this is in the final instance it will look like this: 

![image](https://github.com/user-attachments/assets/604d8830-9817-4c37-b17a-402d2d2869e7)

Reply:
As an extension developer, the 2nd option (compiler as an array) is way better, right? Template literals are prone to errors.

But the result... yeah, the `...[]` looks weird.

Could we get the best of the 2 worlds?

Like if we force `compilers` to be an array... couldn't we just do `compilers = ${massagedCompilers}`? 

(I might be missing something, didn't test... speaking out my mind haha)

Reply:
> Probably if we can omit solidityVersion variable if we have something in compilers, we can use just

Ohh yes we totally remove `solidityVersion` we had to keep it backward compatible but its fine I think if we remove it. I like the solution here but this is only particular this case right? 

I think what Carlos mentioned having best of both world is the way to go I feel. 

Probably some prettier might removing that redundant spread operator or maybe there might some JS library from which we could take inspiration to solve this. 

Also maybe @rin-st lets solve this first and then merge this PR, #125 , #131 ? Because after we make change again we need to update those corresponding .template.mjs files

Reply:
> Ohh yes we totally remove solidityVersion we had to keep it backward compatible but its fine I think if we remove it.

Updated 702063e

> I like the solution here but this is only particular this case right?

It will work similar in all the cases when we need to replace array. In cases we need to add new array values it seems we will need to add that `...[ ]` thing